### PR TITLE
Image component.

### PR DESCRIPTION
**Pull request by kingga**
Added image component by using the canvas component.

```php
<?php

require_once 'vendor/autoload.php';

use Gui\Application;
use Gui\Components\Image;

$app = new Application;

$app->on('start', function () use ($app) {
    $file = sprintf('%s/images/square.png', dirname(__FILE__));
    $imgw = $imgh = 100;

    $image = new Image([
        'top'       => 10,
        'left'      => 10,
        'width'     => $imgw,
        'height'    => $imgh,
    ]);

    $image->setSize($imgw, $imgh)
        ->setFile($file);
});

$app->run();
```

### CODE REVIEW

**Comment by kingga** [comment]
Revising this now using TImage as larger images can take time to load but I'm just figuring out Pascal & Lazarus right now.

File: test/Components/ImageTest.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+class ImageTest extends TestCase
+{
+    public function testMissingFile()
+    {
+        $this->setExpectedException(
+            \Exception::class,
+            'The file could not be found.'
+        );
+        
+        $image = new Image(array(), null, new Application());
```
Comment (reisraff) [inline test/Components/ImageTest.php]: can you use [] instead of array()

File: src/Components/Image.php [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+    {
+        foreach ($this->tmpFiles as $file) {
+            try {
+                fclose($file);
+            } catch (\Throwable $e) {
+                // ...
+            }
+        }
+
+        $this->tmpFiles = array();
```
Comment (reisraff) [inline src/Components/Image.php]: can you use [] instead of array()

File: src/Components/Image.php [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+    }
+
+    /**
+     * Returns the images width and height.
+     *
+     * @return array
+     */
+    public function getSize()
+    {
+        return array(
```
Comment (reisraff) [inline src/Components/Image.php]: can you use [] instead of array()

File: src/Components/Image.php [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+     */
+    private $imgH = null;
+
+    /**
+     * A cache of the images data if loaded before the app
+     * is ready.
+     *
+     * @var array
+     */
+    private $imgData = array();
```
Comment (reisraff) [inline src/Components/Image.php]: can you use [] instead of array()

File: src/Components/Image.php [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+     */
+    protected $imgFile = null;
+
+    /**
+     * Stores all of the temporary files which have been
+     * modified, e.g. resized.
+     *
+     * @var array
+     */
+    private $tmpFiles = array();
```
Comment (reisraff) [inline src/Components/Image.php]: can you use [] instead of array()

File: test/Components/ImageTest.php (hunk 1 above)
Comment (kingga) [reply test/Components/ImageTest.php]: Sorry, originally I did but I wasn't sure if PHP 5 was compatible, I can do that for you. Can I also use type hints for arguments and return values?

**Pull request merged**